tests/update_state_test.py

```diff
@@ -0,0 +1,202 @@
+        game = Game()
+        # Create a code_number to see the change that should happen
+        game.code_number = 3
+        # Create a correct guess for the correct team
+        game.guess = [card.word for card in game.board.cards if card.color == game.team][0]
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is False
+
+        game.check_guess()
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
+        assert game.guess_value == 1
+        assert game.code_number == 2
+
+    def test_guessed_opponents_word(self):
+        'Test if flipped starts as false and ends as true.'
+        'Also check if guess is the other teams and code_number is reset'
+        game = Game()
+
+        game.team = "red"
+        game.code_number = 3
+        # game.guess is the first word that has the color of the opposite teams word
+        game.guess = [card.word for card in game.board.cards if card.color == "blue"][0]
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is False
+
+        game.check_guess()
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
+        assert game.guess_value == 2
+        assert game.code_number == 0
+
+    def test_guessed_neutral_word(self):
+        'Test if flipped starts as false and ends as true.'
+        'Also check if guess is neutral and code_number is reset'
+        # game.guess is the first neutral word
+        game = Game()
+        game.code_number = 3
+        # game.guess is the first word that is a neutral word
+        game.guess = [card.word for card in game.board.cards if card.color == "yellow"][0]
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is False
+
+        game.check_guess()
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
+        assert game.guess_value == 3
+        assert game.code_number == 0
+
+    def test_guessed_death_word(self):
+        'Test if flipped starts as false and ends as true.'
+        'Also check if guess is the death word and code_number is reset'
+        game = Game()
+        game.code_number = 3
+        # game.guess is the first word that is the death word
+        game.guess = [card.word for card in game.board.cards if card.color == "black"][0]
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is False
+
+        game.check_guess()
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
+        assert game.guess_value == 4
+        assert game.code_number == 0
+
+
+class TestCheckWin:
+    '''Check_win is responsible for assigning a winner if the death card 
+    is picked or if the entire deck has been flipped'''
+    def test_winner_and_game_over_for_death_card(self):
```

Comment:
`test_winner_and_game_over_for_death_card` and `test_winner_for_death_card` might be better as `test_red_wins_if_blue_draws_death` and `test_blue_wins_if_red_draws_death`.